ReactAndroid/src/main/java/com/facebook/react/views/art/ARTShapeShadowNode.java

```diff
@@ -182,6 +184,34 @@ protected boolean setupFillPaint(Paint paint, float opacity) {
               (int) (mFillColor[2] * 255),
               (int) (mFillColor[3] * 255));
           break;
+        case 1: // linear gradient
+          if (mFillColor.length < 5) {
+            FLog.w(ReactConstants.TAG, 
+              "[ARTShapeShadowNode setupFillPaint] expects 5 elements, received " 
+              + mFillColor.length);
+            return false;
+          }
+          float x0 = mFillColor[1];
+          float y0 = mFillColor[2];
+          float x1 = mFillColor[3];
+          float y1 = mFillColor[4];
```

Comment:
What are these 4 variables? Add a comment explaining what they represent. Are they the coordinates of the start and end of the gradient? If yes, why not name them `gradientStartX` etc.?